www/src/context/AlertContext/SnackBarProvider.tsx

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, createContext, useContext } from 'react';
+import Snackbar from '@material-ui/core/Snackbar';
+import { SnackBarType } from './SnackBarType';
+
+export const snackBarContext = createContext<SnackBarType>({message: '', updateMessage: () => {}});
+export const SnackBarProvider = ({
+    children,
+}: JSX.ElementChildrenAttribute): JSX.Element => {
+    const [message, setMessage] = useState('')
+    const updateMessage = (m:string) => {
+        setMessage(m);
+        setOpen(true);
+    }
+    const [open, setOpen] = useState(true)
+
+    useEffect(() => {
+        if (message.length > 0 && open) {
+            const timer = setTimeout(() => setOpen(false), 5000)
+            return () => clearTimeout(timer)
+        }
+    })
+
+    return (
+        <snackBarContext.Provider value={{ message, updateMessage }}>
+            {children} 
+            {message.length > 0 &&
+            <Snackbar open={open} anchorOrigin={{ vertical: 'top', horizontal: 'center' }} message={message} />
+            }
+        </snackBarContext.Provider>
+    )
+}
```

Comment:
See AuthContext.tsx to see how to create a singleton context and export that as a hook.